include/swift/AST/LocalizationFormat.h

```diff
@@ -0,0 +1,99 @@
+//===--- LocalizationFormat.h - YAML format for Diagnostic Messages ---*-
+// C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2020 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+// This file defines the format for localized diagnostic messages.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef SWIFT_LOCALIZATIONFORMAT_H
+#define SWIFT_LOCALIZATIONFORMAT_H
+
+#include "swift/AST/DiagnosticsCommon.h"
+#include "llvm/Support/YAMLParser.h"
+#include "llvm/Support/YAMLTraits.h"
+
+enum class swift::DiagID : uint32_t {
```

Comment:
I don't think this is correct because localization doesn't own diagnostics it only references them. Can't this be forward declared and `ScalarEnumerationTraits`  and `MappingTraits` moved to `LocalizationFormat.cpp` which could be loaded after `DiagnosticList.cpp` in cmake file?